### PR TITLE
nixos-rebuild: allow to override builders

### PR DESCRIPTION
**Pull request by Mic92**
Since nix 2.0 the no-build-hook option was replaced by the builders options
that allows to override remote builders ad-hoc.
Since it is useful to disable remote builders updating nixos without network,
this commit reintroduces the option.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: nixos/doc/manual/man-nixos-rebuild.xml [hunk 1]
```diff
@@ -315,6 +319,27 @@ $ ./result/bin/run-*-vm
...
+       <replaceable>builder-spec</replaceable>
+     </term>
+     <listitem>
+       <para>
+         Allow to specify remote builders ad-hoc for building the new system.
+         This requires user executing <command>nixos-rebuild</command> (usually
+         root) to be configured as trusted user in nix daemon. This can be
+         achived by using the <literal>nix.trustedUsers</literal> nixos option.
+         Examples values for that option are described in the
+         <literal>Remote builds chapter<literal> in the Nix manual,
```
Comment (Mic92) [inline nixos/doc/manual/man-nixos-rebuild.xml]: @grahamc can you help me out with converting this to a cross-reference here?

File: nixos/doc/manual/man-nixos-rebuild.xml [hunk 2]
```diff
@@ -315,6 +319,27 @@ $ ./result/bin/run-*-vm
...
+     <term>
+       <option>--builders</option>
+       <replaceable>builder-spec</replaceable>
+     </term>
+     <listitem>
+       <para>
+         Allow to specify remote builders ad-hoc for building the new system.
+         This requires user executing <command>nixos-rebuild</command> (usually
+         root) to be configured as trusted user in nix daemon. This can be
+         achived by using the <literal>nix.trustedUsers</literal> nixos option.
```
Comment (Mic92) [inline nixos/doc/manual/man-nixos-rebuild.xml]: nixos options probably also have a special markup?

File: nixos/modules/installer/tools/nixos-rebuild.sh [hunk 3]
```diff
@@ -53,11 +53,11 @@ while [ "$#" -gt 0 ]; do
         repair=1
         extraBuildFlags+=("$i")
         ;;
-      --max-jobs|-j|--cores|-I)
+      --max-jobs|-j|--cores|-I|--builders)
         j="$1"; shift 1
         extraBuildFlags+=("$i" "$j")
         ;;
-      --show-trace|--no-build-hook|--keep-failed|-K|--keep-going|-k|--verbose|-v|-vv|-vvv|-vvvv|-vvvvv|--fallback|--repair|--no-build-output|-Q|-j*)
```
Comment (Mic92) [inline nixos/modules/installer/tools/nixos-rebuild.sh]: `--no-build-hook` is no longer accepted since nix 2.0.

**Comment by Mic92** [comment]
There was also an issue related to that. @vcuncat do you remember which one this was?

File: nixos/doc/manual/man-nixos-rebuild.xml (hunk 2 above)
Comment (edolstra) [reply nixos/doc/manual/man-nixos-rebuild.xml]: `as trusted user in nix daemon` -> `as a trusted user in the Nix daemon`.

`nixos option` -> `NixOS option`.

File: nixos/doc/manual/man-nixos-rebuild.xml [hunk 4]
```diff
@@ -315,6 +319,27 @@ $ ./result/bin/run-*-vm
...
    </varlistentry>
+   <varlistentry>
+     <term>
+       <option>--builders</option>
+       <replaceable>builder-spec</replaceable>
+     </term>
+     <listitem>
+       <para>
+         Allow to specify remote builders ad-hoc for building the new system.
+         This requires user executing <command>nixos-rebuild</command> (usually
```
Comment (edolstra) [inline nixos/doc/manual/man-nixos-rebuild.xml]: `user` -> `the user`.

File: nixos/doc/manual/man-nixos-rebuild.xml [hunk 5]
```diff
@@ -315,6 +319,27 @@ $ ./result/bin/run-*-vm
...
     </listitem>
    </varlistentry>
+   <varlistentry>
+     <term>
+       <option>--builders</option>
+       <replaceable>builder-spec</replaceable>
+     </term>
+     <listitem>
+       <para>
+         Allow to specify remote builders ad-hoc for building the new system.
```
Comment (FRidh) [inline nixos/doc/manual/man-nixos-rebuild.xml]:
```suggestion
         Allow ad-hoc remote builders for building the new system.
```

File: nixos/doc/manual/man-nixos-rebuild.xml [hunk 6]
```diff
@@ -315,6 +319,27 @@ $ ./result/bin/run-*-vm
...
+         This requires the user executing <command>nixos-rebuild</command> (usually
+         root) to be configured as a trusted user in the Nix daemon. This can be
+         achived by using the <literal>nix.trustedUsers</literal> NixOS option.
```
Comment (FRidh) [inline nixos/doc/manual/man-nixos-rebuild.xml]:
```suggestion
         achieved by using the <literal>nix.trustedUsers</literal> NixOS option.
```

**Comment by dtzWill** [comment]
`--option builders ""` already works, FWIW? Does this add support for the more direct `--builders ""`?

**Comment by LnL7** [comment]
Ah indeed, that explains my confusion.

**Comment by Mic92** [comment]
@dtzWill yes, plus some documentation, which does not hurt.

**Comment by dtzWill** [comment]
Yes indeed, thank you! \o/

**Comment by vcunat** [comment]
This merge broke the build of NixOS manual: https://hydra.nixos.org/build/87335696

**Pull request merged**